libgnucash/engine/gnc-budget.cpp

```diff
@@ -471,31 +496,46 @@ gnc_budget_get_num_periods(const GncBudget* budget)
     return GET_PRIVATE(budget)->num_periods;
 }
 
-static inline void
-make_period_path (const Account *account, guint period_num, char *path1, char *path2)
+static inline StringVec
+make_period_data_path (const Account *account, guint period_num)
 {
-    const GncGUID *guid;
-    gchar *bufend;
-    guid = xaccAccountGetGUID (account);
-    guid_to_string_buff (guid, path1);
-    g_sprintf (path2, "%d", period_num);
+    gchar guid_buf [GUID_ENCODING_LENGTH + 1];
+    guid_to_string_buff (xaccAccountGetGUID (account), guid_buf);
+    std::string acct_guid { guid_buf };
+    std::string period_str { std::to_string (period_num) };
+    return { acct_guid, period_str };
 }
 
+static inline StringVec
+make_period_note_path (const Account *account, guint period_num)
+{
+    StringVec path { GNC_BUDGET_NOTES_PATH };
+    StringVec data_path { make_period_data_path (account, period_num) };
+    std::move (data_path.begin(), data_path.end(), std::back_inserter (path));
```

Comment:
std::move doesn't do anything here.

Reply:
It's a different move: https://en.cppreference.com/w/cpp/algorithm/move because period_data_path is `{guid, period_num}`, and period_note_path is `{guid, period_num, "notes"}`.

Reply:
Oh, cool. Nice find.